Make search button sticky

esoergel opened the pull request:
## Product Description

I stuck the search buttons on a white background with a little border and made that sticky:

![image](https://github.com/user-attachments/assets/9d2610aa-85c9-49c2-95f3-5fffec42c1c4)

![image](https://github.com/user-attachments/assets/bd55dff6-e4cb-4c42-bb2f-b74d98fb1182)

In mobile mode:
![image](https://github.com/user-attachments/assets/02c75abb-ce5e-4b26-a5f7-f61381a8eb6a)

## Technical Summary
<!--
    Provide a link to the ticket or document which prompted this change,
    Describe the rationale and design decisions.
-->
https://dimagi.atlassian.net/browse/USH-4682

The second commit was added to address a specific edge case.  If the main content is much shorter than the list of filters, you can scroll entirely past it, which means the breadcrumbs disappear, making the position no longer appropriate:
![image](https://github.com/user-attachments/assets/5f9934bd-e3a8-4aec-88c8-333a37df84e2)

I think the problem was that the sidebar overflows the main container because it uses `position: absolute;` rather than flexbox or similar to display in a column next to the rest of it.  The DOM is structured like this:
```
div #menu-container
  div #breadcrumb-region
  div #content-container
    div #sidebar-region - contains the search sidebar, has `position: absolute`
    div #menu-region - contains the case list and the main body of the page.  Has a left margin to leave space for the sidebar
```

I addressed it by using flexbox columns there instead, and it seems to work well, though it needs more testing.

## Feature Flag
Split screen case search

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->
https://dimagi.atlassian.net/browse/QA-7019

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

esoergel committed:
Make query search buttons sticky

Tricky part here was making sure the buttons don't overlap the
breadcrumbs when present, but go to the top of the screen when they
aren't.  I couldn't figure out a good way to do this by structuring the
DOM properly, so I decided to set a class on the containing element
indicating whether the breadcrumbs were present, then use this for the
CSS positioning selector.

esoergel committed:
Use flexbox for columns in main webapps content

The old behavior was to conditionally offset the "main" section to allow
room for the sidebar, and then put the sidebar on top of that offset
margin, with position: absolute.  This made the sidebar overflow the
div, which messed up the sticky scrolling effect.

We're also planning to soon make the persistent menu and case search map
toggleable, and I think using flex containers will make that all a bit
easier.

AddisonDunn reviewed (commented):
From [the docs on sticky](https://developer.mozilla.org/en-US/docs/Web/CSS/position):

> The element is positioned according to the normal flow of the document, and then offset relative to its nearest scrolling ancestor

This makes me think the buttons should be entirely dependent on the "scrolling ancestors", i.e. the breadcrumbs bar and the header. I think it's a great change to move to the flexbox, but I'm confused how that would fix the particular bug you mentioned.

esoergel commented:
> The element is positioned according to the normal flow of the document, and then offset relative to its nearest scrolling ancestor

> This makes me think the buttons should be entirely dependent on the "scrolling ancestors", i.e. the breadcrumbs bar and the header. I think it's a great change to move to the flexbox, but I'm confused how that would fix the particular bug you mentioned.

@AddisonDunn the problem was that the sidebar region was marked `position: absolute`, which removes it from the nromal document flow.  This meant it was able to overflow its containing div:

![image](https://github.com/user-attachments/assets/80273e94-b488-4ab8-85a9-8d906a5b48f9)

The containing div stops just above the "Formplayer Version: 2.53, App Version: 31" text, and if you look in the DOM inspector, you'll see Firefox has helpfully tagged it `overflow`.  This meant that the new sticky buttons could scroll below the div that contains the `sidebar-region` and the breadcrumbs bar but still be within the `sidebar-region` div.

esoergel commented on corehq/apps/cloudcare/templates/cloudcare/formplayer_home.html:
```diff
@@ -107,10 +107,10 @@
         <div id="sidebar-region" class="noprint-sub-container"></div>
-        <div id="menu-region" class="print-container"></div>
-        <section id="webforms" data-bind="
+        <div id="menu-region" class="print-container flex-grow-if-not-empty"></div>
```
Yeah, the map is somewhere inside the `menu-region`, so it will take up part of the space allotted to this div.

esoergel committed:
Fix non-split-screen case search

I'd been overzealous in originally changing this from #sidebar-region
to #sidebar-contents.  The same template is used in non-split-screen
case search, and these styles shouldn't apply there.

AddisonDunn reviewed (approved):
@esoergel OK, I think I finally get what's going on. Thanks for explaining.

esoergel commented:
I removed the bootstrap diffs from `master`, leaving this branch with a conflict, so I rebased and dropped the two bootstrap diff commits.
